### PR TITLE
restructure priv risk score to be mia on point basis #151

### PR DESCRIPTION
**Pull request by blauertee**
Wasn't able to test thoroughly again because of cuda errors, so please run the code :)

### CODE REVIEW

File: privacy_evaluator/attacks/membership_inference/on_point_basis.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+        :param num_bins: the number of bins used to compute the training/test histogram
+        :return: membership probability results
+        """
+        loss_train = self.target_model.art_classifier.compute_loss(self.x_train, self.y_train)
+        loss_test = self.target_model.art_classifier.compute_loss(self.x_test, self.y_test)
+        return self._compute_membership_probability(loss_train, loss_test, num_bins)
+
+
+    def _compute_membership_probability(
```
Comment (marisanest) [inline privacy_evaluator/attacks/membership_inference/on_point_basis.py:73]: You could have left this functin in the metrics module. But its only used here so it's fine.

**Review by marisanest — CHANGES_REQUESTED** [review]
Nice implementation :100: I fixed black formatting and a little bug. Please also check the notebooks. I think you also need to change them as well.

**Review by budmil — APPROVED** [review]

**Comment by Erdnaf** [comment]
hey, I tried to run the new notebook, but when running it on my PC my pc stragely hangs on the cell where the attack is performed and frezzes 😢 . When running it on colab I get an Cuda error. Are you shure that it runs for you? Also in the notebook?
@blauertee @marisanest @jtorhoff

**Comment by jtorhoff** [comment]
I think we have a little problem here:



The process RAM usage keeps exploding. Maybe we should wait with this PR for now any then try to get to the root of the problem, but it surely is unexpected.

**Comment by jtorhoff** [comment]
Btw it looks like it behaved the same way before, it's just that we didn't calculate the score for the whole dataset but only first 100 data points. Just tried it out as it was before this PR and the same problem with RAM usage occurs.

**Pull request merged**